Source/Core/DolphinQt/NetPlay/NetPlayDialog.cpp

```diff
@@ -458,6 +462,57 @@ void NetPlayDialog::show(std::string nickname, bool use_traversal)
   UpdateGUI();
 }
 
+void NetPlayDialog::UpdateDiscordPresence()
+{
+#ifdef USE_DISCORD_PRESENCE
+  // both m_current_game and m_player_count need to be set for the status to be displayed correctly
+  if (m_player_count == 0 || m_current_game.empty())
+    return;
+
+  const auto use_default = [this]() {
+    Discord::UpdateDiscordPresence(m_player_count, Discord::SecretType::Empty, "", m_current_game);
+  };
+
+  if (g_TraversalClient)
+  {
+    const auto host_id = g_TraversalClient->GetHostID();
+    if (host_id == decltype(host_id)())
+      return use_default();
+
+    Discord::UpdateDiscordPresence(m_player_count, Discord::SecretType::RoomID,
+                                   std::string(host_id.begin(), host_id.end()), m_current_game);
+  }
+  else if (IsHosting())
+  {
+    if (m_exernal_ip_address.empty())
+    {
+      Common::HttpRequest request;
+      // ENet does not support IPv6, so IPv4 has to be used
+      request.UseIPv4();
+      Common::HttpRequest::Response response =
+          request.Get("https://ip.dolphin-emu.org/", {{"X-Is-Dolphin", "1"}});
+
+      if (!response.has_value())
+        return use_default();
+      m_exernal_ip_address = std::string(response->begin(), response->end());
+    }
+    const int port = Settings::Instance().GetNetPlayServer()->GetPort();
+
+    Discord::UpdateDiscordPresence(m_player_count, Discord::SecretType::IPAddress,
+                                   Discord::CreateSecretFromIPAddress(m_exernal_ip_address, port),
+                                   m_current_game);
+  }
+  else
+  {
+    Discord::UpdateDiscordPresence(
```

Comment:
This call doesn't really make sense, as we're not the host. Should probably just be `use_default()`.